packages/mephisto-task/src/live.js

```diff
@@ -24,27 +24,67 @@ const AGENT_STATUS = {
   MEPHISTO_DISCONNECT: "mephisto disconnect",
 };
 
+const STATUS_TO_TEXT_MAP = {
+  [AGENT_STATUS.EXPIRED]:
+    "This task is no longer available to be completed. " +
+    " Please return it and try a different task",
+  [AGENT_STATUS.TIMEOUT]:
+    "You took to long to respond to this task, and have timed out. " +
+    "The task is no longer available, please return it.",
+  [AGENT_STATUS.DISCONNECT]:
+    "You have disconnected from our server during the duration of the task. " +
+    "If you have done substantial work, please reach out to see if we can recover it. ",
+  [AGENT_STATUS.RETURNED]:
+    "You have disconnected from our server during the duration of the task. " +
+    "If you have done substantial work, please reach out to see if we can recover it. ",
+  [AGENT_STATUS.PARTNER_DISCONNECT]:
+    "One of your partners has disconnected while working on this task. We won't penalize " +
+    "you for them leaving, so please submit this task as is.",
+  [AGENT_STATUS.MEPHISTO_DISCONNECT]:
+    "Our server appears to have gone down during the " +
+    "duration of this Task. Please send us a message if you've done " +
+    "substantial work and we can find out if the task is complete enough to " +
+    "compensate.",
+};
+
 const useMephistoLiveTask = function (props) {
   const [connectionStatus, setConnectionStatus] = React.useState("");
-  const [agentState, setAgentState] = React.useState(null);
   const [agentStatus, setAgentStatus] = React.useState(null);
 
   const defaultConnectionHook = useMephistoSocket;
   const useConnectionHook = props.customConnectionHook || defaultConnectionHook;
 
   const taskProps = useMephistoTask();
+
+  const { initialTaskData } = taskProps;
+
+  function onMessageReceived(message) {
+    const { state, message_id, timestamp, ...remaining_message } = message;
+    if (state !== undefined) {
+      props.onStateUpdate && props.onStateUpdate({ state });
```

Comment:
I thought we got rid of state and are using statuses? Is this a typo or am I missing something?

Reply:
So this was noted a bit in the PR description:
> One element that remains a bit leaky - triggering these kinds of state callbacks requires that the live data packet you send contains a { 'state': {...}} field...

I imagine that, while we're dropping `state` as an official field that we track in the package, I expect that downstream tasks may still want to use `state` for their local task context. That type of `state` under the hood is really just a regular message, rather than being something akin to status which is Mephisto-specific. Here I'm exposing a way for Mephisto to handle splitting these out such that, to the protocol, there are only `liveData` messages, but to a `mephisto-task` user they can register separate `state` and `message` callbacks. 

If this is undesirable, I can push this further along and just have `bootstrap-chat`'s message function do this split instead.

Reply:
I'm still not sold on us first-classing state as part of the library, and think it may be better for users to implement that and choose to use `liveData` in that way if they wish.